### PR TITLE
Detach address

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/ParcelRegistry.Api.BackOffice.Handlers.Lambda/Handlers/AttachAddressHandler.cs [hunk 1]
```diff
@@ -64,6 +64,7 @@ protected override async Task<ETagResponse> InnerHandle(AttachAddressLambdaReque
         {
             return exception switch
             {
+                ParcelIsRemovedException => ValidationErrors.Common.ParcelRemoved.ToTicketError,
```
Comment (jvandaal) [inline src/ParcelRegistry.Api.BackOffice.Handlers.Lambda/Handlers/AttachAddressHandler.cs:67]: Can ParcelIsRemovedException be moved to the base ParcelLambdaHandler?

File: src/ParcelRegistry.Api.BackOffice.Handlers.Lambda/Requests/DetachAddressLambdaRequest.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{
+    using Abstractions.Requests;
+    using Abstractions.SqsRequests;
+    using Be.Vlaanderen.Basisregisters.Sqs.Lambda.Requests;
+    using Parcel;
+    using Parcel.Commands;
+
+    public sealed record DetachAddressLambdaRequest : SqsLambdaRequest, IHasParcelId
+    {
+        public DetachAddressRequest Request { get; set; }
```
Comment (jvandaal) [inline src/ParcelRegistry.Api.BackOffice.Handlers.Lambda/Requests/DetachAddressLambdaRequest.cs:11]: Change to get; only?

File: src/ParcelRegistry.Api.BackOffice.Handlers.Lambda/Handlers/AttachAddressHandler.cs (hunk 1 above)
Comment (ArneD) [reply src/ParcelRegistry.Api.BackOffice.Handlers.Lambda/Handlers/AttachAddressHandler.cs:67]: It is marked as resolved, but the exception is still here. Did you have any discussion? If so please add it here :)

File: src/ParcelRegistry.Api.BackOffice/Handlers/DetachAddressHandler.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        protected override IDictionary<string, string> WithTicketMetadata(string aggregateId, DetachAddressSqsRequest sqsRequest)
+        {
+            return new Dictionary<string, string>
+            {
+                { RegistryKey, nameof(ParcelRegistry) },
+                { ActionKey, Action },
+                { AggregateIdKey, aggregateId },
+                { ObjectIdKey, aggregateId }
```
Comment (ArneD) [inline src/ParcelRegistry.Api.BackOffice/Handlers/DetachAddressHandler.cs:28]: The capakey is the objectid. Probably wrong in attach handler as well.

File: src/ParcelRegistry/Parcel/Parcel.cs [hunk 4]
```diff
@@ -70,6 +72,38 @@ public void AttachAddress(AddressPersistentLocalId addressPersistentLocalId)
...
+
+            if (address is null)
+            {
+                throw new AddressNotFoundException();
+            }
+
+            if (address.Value.IsRemoved)
+            {
+                throw new AddressIsRemovedException();
+            }
```
Comment (ArneD) [inline src/ParcelRegistry/Parcel/Parcel.cs:89]: Shouldn't this be idempotent?
If it is removed, then it shouldn't be here in the first place so you can accept the request to detach.

And should `IsNotFound` check not be on `AddressPersistentLocalIds` ?

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/ParcelRegistry.Api.BackOffice.Handlers.Lambda/Handlers/AttachAddressHandler.cs (hunk 1 above)
Comment (pgallik) [reply src/ParcelRegistry.Api.BackOffice.Handlers.Lambda/Handlers/AttachAddressHandler.cs:67]: I thought this comment was about the detach handler where I had removed it, will do here as well.

**Review by ArneD — APPROVED** [review]

**Pull request merged**